### ISSUE TITLE
Implement `Series.cat.categories` and `Series.cat.ordered`

### ISSUE THREAD

**Issue by stinodego** [pola-rs/polars#5671]
### Problem description

Required for #5662 

We should be able to get the categories of a Categorical Series, and get whether the ordering is lexical or physical.

If I understand correctly, a Series of type `Categorical` is really a Series of type `UInt32` with a mapping of a `UInt32` value to a string. We should be able to access this mapping.

I propose we access this through a `@property` in the `Series.cat` namespace, with the result being a `Series` of type `Utf8` (could also be a `list[str]`, but that's probably not as efficient), where the index is the `UInt32` (physical) value.

```python
s = pl.Series(['a', 'c', 'b', 'b'], dtype=pl.Categorical)
s.cat.categories  # new property
# shape: (3,)
# Series: '' [str]
# [
#        "a"
#        "c"
#        "b"
# ]
```

By default, the ordering is physical:

```python
s.sort()
# shape: (4,)
# Series: '' [cat]
# [
#         "a"
#         "c"
#         "b"
#         "b"
# ]
```

We can set the ordering to lexical, but there is no way to find if the Series is currently ordered lexically or physically. I propose we add a `@property` to the `Series.cat` namespace that indicates whether or not the Series is ordered lexically.

```python
s.cat.ordered  # new property, returns False
s = s.cat.set_ordering('lexical')
s.cat.ordered  # returns True

s.sort()
# shape: (4,)
# Series: '' [cat]
# [
#         "a"
#         "b"
#         "b"
#         "c"
# ]

I am not sure what happens here under the hood, but I imagine calling `set_ordering` affects the mapping of the `UInt32` value to string. In that case, it should affect the `Series.cat.categories` property:

```python
s.cat.categories
# shape: (3,)
# Series: '' [str]
# [
#        "a"
#        "b"
#        "c"
# ]
```

So, in short, the `CatNameSpace` class should get two new properties with the following signatures:

```python
class CatNameSpace

    @property
    def ordered(self) -> bool:
        """Returns whether the Categorical is ordered lexically."""
        ...

    @property
    def categories(self) -> pl.Series:
        """Returns the categories of the Categorical as a Series."""
        ...
```

**Comment by ritchie46** [comment]
The ` categories` is already possible without any rust side changes:

```python
self.to_physical() 
```

Gives the physical representation, e.g. the `u32` backed categorical integers.

**Comment by mcrumiller** [comment]
@ritchie46 those aren't the categories though, those are the indexes, right? As far as I know, you have to do `s.unique().cast(pl.Utf8)` to get the categories.

I like the idea of `.cat.categories` returning a series of the categories and `.cat.ordering` returning "lexical" or "physical".

A `.cat.set_order(categories)` would be nice too (with physical ordering), but categoricals can be somewhat of a rabbit hole and until someone actually needs this probably is not worth the time.

**Comment by stinodego** [comment]
Indeed, `to_physical` isn't enough. I think we need to somehow access that `rev_map`.

It seems the `.cat.ordered` property is available on the Rust side as `CategoricalChunked.use_lexical_sort`.

**Comment by ritchie46** [comment]
> Indeed, `to_physical` isn't enough. I think we need to somehow access that `rev_map`.
> 
> It seems the `.cat.ordered` property is available on the Rust side as `CategoricalChunked.use_lexical_sort`.

A `cat_s.unique().to_physical()` is internally optimized to give access to the `RevMap`, so we could use that.

**Comment by stinodego** [comment]
> A `cat_s.unique().to_physical()` is internally optimized to give access to the `RevMap`, so we could use that.

Great, that works perfectly 👍
